inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -920,3 +921,81 @@ def normalize_collaborations(obj, eng):
         collaboration['value'] = collaboration_normalized_name
         collaboration['record'] = response[0].self.to_dict()
     return obj
+
+
+def _match_lit_author_affiliation(raw_aff):
+    query = Q(
+        Q(
+            "nested",
+            path="authors",
+            query=(
+                Q("match", authors__raw_affiliations__value=raw_aff)
+                & Q("exists", field="authors.affiliations.value")
+            ),
+            inner_hits={},
+        )
+        & Q("term", _collections="Literature")
+        & Q("term", curated=True)
+    )
+    result = (
+        Search(index="records-hep", using=current_search_client)
+        .query(query)
+        .execute()
+        .hits
+    )
+    if result:
+        matched_author = result[0].meta.inner_hits.authors.hits[0].to_dict()
+        return matched_author["affiliations"][0]
+
+
+def _match_institution(matched_affiliation):
+    query = Q("match", legacy_ICN=matched_affiliation["value"])
+    result = InstitutionsSearch().query(query).execute()
+    if result:
+        return result.hits[0]["self"]
+
+
+def _assign_institution_ref_to_affiliation(matched_affiliation):
+    institution_reference = _match_institution(matched_affiliation)
+    if institution_reference:
+        matched_affiliation["record"] = institution_reference
+    return matched_affiliation
+
+
+def _assign_matched_affiliation_to_author(
+    author_affiliations, matched_author_affiliation
+):
+    if "record" not in matched_author_affiliation:
+        matched_author_affiliation = _assign_institution_ref_to_affiliation(
+            matched_author_affiliation
+        )
+    author_affiliations.append(matched_author_affiliation)
+    return matched_author_affiliation
+
+
+def normalize_affiliations(obj, eng):
```

Comment:
does this ever get called? As far as I can tell, this creates the workflow step but doesn't use it anywhere.